src/hdmf_zarr/backend.py

```diff
@@ -426,6 +434,61 @@ def write_builder(self, **kwargs):
         self.logger.debug("Done writing %s '%s' to path '%s'" %
                           (f_builder.__class__.__qualname__, f_builder.name, self.source))
 
+        # Consolidate metadata for the entire file after everything has been written
+        zarr.consolidate_metadata(store=self.path)
+
+    def consolidate_metadata(self, store):
+        """
+        When a file is written, the metadata within the file is consolidated automatically.
+        If there are any metadata changes, the user needs to consolidate the metadata again
+        with this method in order for the metadata to be read correctly.
+
+        Consolidate all metadata for groups and arrays within the given store into a
+        single resource and put it under .zmetadata.
+        """
+        zarr.consolidate_metadata(store, metadata_key='.zmetadata')
+
+    def __get_store_path(self, zarr_object):
```

Comment:
Instead changing `get_zarr_parent_path` and  `get_zarr_paths`to non-static method, I think we can simply make this a static method. It looks like the `self` argument is not needed here, so making this function static should be fine. 

```suggestion
    @staticmethod
    def __get_store_path(zarr_object):
```

Reply:
Yep already did that locally.